tornadox/blocks/step_impl/first_order/ek1_projmatfree_d_nu.py

```diff
@@ -0,0 +1,147 @@
+"""Projection-matrix-free version of the EK1."""
+
+
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+
+from tornadox.blocks import sqrtutil
+
+
+@partial(jax.jit, static_argnames=("f", "df", "num_derivatives"))
+def attempt_step_forward_only(
```

Comment:
The "forward only" was quite unclear to me. Maybe we could make it more explicit in the docstring that there is the other method that precomputes some smoothing stuff? 

...on the other hand, we'll probably polish docstrings later, so I'm also fine with leaving it as it is.

Reply:
Yeah probably. The "forward_only" kind of means "no_dense_output" but it is technically not restricted to dense-output, but to anything smoothing-distribution-related. 

How about we do this kind of polishing later as things bother us?

Reply:
On this regard, I'm kind of tempted now to merge the two functions, there is so much duplication... But I know that that's not really the spirit of this project :sweat_smile:

Reply:
No!!! :smile: